src/screens/tabs/Tokens/components/AddToken/steps/ReviewToken/index.tsx

```diff
@@ -22,11 +23,28 @@ export function ReviewToken({ token, onClose }: Props) {
   const { principal } = useSelector(state => state.keyring?.currentWallet);
   const { icpPrice } = useSelector(state => state.icp);
   const [balance, setBalance] = useState<{ amount: number; value?: number }>();
+  const [error, setError] = useState<string>();
   const [loadingBalance, setLoadingBalance] = useState(true);
   const [loadingRegister, setLoadingRegister] = useState(false);
 
   const dispatch = useDispatch();
 
+  function renderError() {
+    return (
+      <View style={styles.tokenContainer}>
+        <TokenIcon logo={token?.logo} />
+        <View style={styles.errorTextContainer}>
+          <Text type="body1" style={styles.errorTitle}>
+            {token?.name}
+          </Text>
+          <Text type="body1" style={styles.errorSubtitle}>
+            {error}
+          </Text>
+        </View>
+      </View>
+    );
+  }
+
   function renderToken() {
```

Comment:
What if we extract this 2 into components? is there a reason they are here?

Reply:
they'll only be used in this screen. and renderToken is just <TokenItem /> and a loader